### PR TITLE
Feature/drop and sort entry analytical statement

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 1]
```diff
@@ -25,29 +35,73 @@ function AnalyticalEntryInput(props: AnalyticalEntryInputProps) {
...
-            <h4>
-                {value.entry}
-            </h4>
-            <QuickActionButton
-                name={index}
-                onClick={onRemove}
-                // FIXME: use translation
-                title="Remove Analytical Entry"
+            name="entry"
+            onDrop={!entryDraggedStatus ? handleAnalyticalEntryAdd : undefined}
```
Comment (tnagorra) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx:75]: Are we disabling drop for the item that is being dragged?
Maybe we should add a comment for this

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 2]
```diff
@@ -52,35 +61,69 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
...
             }
+
             // NOTE: Don't let users add more that certain entries
-            if ((value.analyticalEntries?.length ?? 0) >= ENTRIES_LIMIT) {
+            if (
+                isNotDefined(movedItem)
+                && (value?.analyticalEntries?.length ?? 0) >= ENTRIES_LIMIT
+            ) {
                 return;
             }
```
Comment (tnagorra) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx]: We should move this before `const newAnalyticalEntries = [...(value.analyticalEntries ?? [])];`

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 3]
```diff
@@ -52,35 +61,69 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
...
-                order: 0,
-            };
+            if (dropOverEntryUuid) {
+                const currentIndex = newAnalyticalEntries
+                    .findIndex(v => v.uuid === dropOverEntryUuid);
+                newAnalyticalEntries.splice(currentIndex, 0, newAnalyticalEntry);
+            } else {
+                newAnalyticalEntries.push(newAnalyticalEntry);
+            }
+            const orderedEntries = newAnalyticalEntries.map((v, i) => ({ ...v, order: i }));
```
Comment (tnagorra) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx]: Please add comment

File: src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx [hunk 4]
```diff
@@ -25,29 +35,73 @@ function AnalyticalEntryInput(props: AnalyticalEntryInputProps) {
...
+                    </p>
+                )}
+                <h4>
+                    {value.entry}
+                </h4>
+                <QuickActionButton
+                    name={index}
+                    onClick={onRemove}
+                    // FIXME: use translation
+                    title="Remove Analytical Entry"
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/AnalyticalEntryInput/index.tsx]: why translations is not used here?

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 5]
```diff
@@ -52,35 +61,69 @@ function AnalyticalStatementInput(props: AnalyticalStatementInputProps) {
         onValueRemove: onAnalyticalEntryRemove,
     } = useFormArray('analyticalEntries', value.analyticalEntries ?? [], onFieldChange);
 
-    const handleAnalyticalEntryAdd = useCallback(
-        (val: Record<string, unknown> | undefined) => {
-            if (!val) {
-                return;
+    const handleAnalyticalEntryDrop = useCallback(
+        (dropValue: DroppedValue, dropOverEntryUuid?: string) => {
+            if (isDefined(dropValue.statementUuid) && (dropValue.statementUuid !== value.uuid)) {
```
Comment (samshara) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx]: value.uuid should be added to dependency array

File: src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx [hunk 6]
```diff
@@ -23,64 +28,117 @@ import AnalyticalEntryInput from './AnalyticalEntryInput';
...
+                'analyticalEntries' as const,
+            );
+            if (isDefined(dropValue.statementUuid) && (dropValue.statementUuid !== value.uuid)) {
+                // NOTE: Remove entry from old statement if it was moved from
+                // one statement to another statement
+                onEntryMove(dropValue.entryId, dropValue.statementUuid);
+            }
+        },
+        [onFieldChange, value.analyticalEntries, value.order, onEntryMove, value.uuid],
+    );
```
Comment (tnagorra) [inline src/views/PillarAnalysis/AnalyticalStatementInput/index.tsx:132]: This looks way more complicated!

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**